engine/util/util_test.go

```diff
@@ -164,3 +168,77 @@ func TestStructTagToFieldName2(t *testing.T) {
 	t.Logf("got output: %+v", m)
 	t.Logf("got error: %+v", err)
 }
+
+func TestStructKindToFieldNameTypeMap(t *testing.T) {
+	k := "test-kind"
+
+	engine.RegisterResource(k, func() engine.Res { return &testEngineRes{} })
+	res, err := StructKindToFieldNameTypeMap(k)
+
+	expected := map[string]*types.Type{
+		"PublicProp1": types.TypeStr,
+		"PublicProp2": types.TypeStr,
+	}
+	require.NoError(t, err)
+	require.Equal(t, expected, res)
+}
+
+type testEngineRes struct {
+	PublicProp1  string
+	PublicProp2  string
+	privateProp1 string
+	privateProp2 string
+}
+
+func (t *testEngineRes) CheckApply(_ bool) (bool, error) {
+	return false, nil
+}
+
+func (t *testEngineRes) Close() error {
+	return nil
+}
+
+func (t *testEngineRes) Cmp(_ engine.Res) error {
+	return nil
+}
+
+func (t *testEngineRes) Default() engine.Res {
+	return t
+}
+
+func (t *testEngineRes) Init(_ *engine.Init) error {
+	return nil
+}
+
+func (t *testEngineRes) Kind() string {
+	return "test-kind"
+}
+
+func (t *testEngineRes) MetaParams() *engine.MetaParams {
+	return nil
+}
+
+func (t *testEngineRes) Name() string {
+	return "test-name"
+}
+
+func (t *testEngineRes) SetKind(_ string) {
+}
+
+func (t *testEngineRes) SetMetaParams(_ *engine.MetaParams) {
```

Comment:
You don't need these underscores since the inputs aren't being used, you can omit them.

Reply:
Didn't know it was possible, thanks!
Fixed in 8ef2386